pkg/executor/bacalhau/bacalhau.go

```diff
@@ -2,115 +2,85 @@ package bacalhau
 	"github.com/lilypad-tech/lilypad/pkg/system"
-	"github.com/rs/zerolog/log"
 )
 
 const RESULTS_DIR = "bacalhau-results"
 
 type BacalhauExecutorOptions struct {
-	ApiHost string
-	ApiPort string
+	ApiHost               string
+	ApiPort               string
+	JobStatusPollInterval uint64
 }
 
 type BacalhauExecutor struct {
 	Options        BacalhauExecutorOptions
 	bacalhauEnv    []string
 	bacalhauClient BacalhauClient
-	ipfsClient     *ipfs.Client
 }
 
-func NewBacalhauExecutor(options BacalhauExecutorOptions, ipfsClient *ipfs.Client) (*BacalhauExecutor, error) {
-	client, err := newClient(options)
+func NewBacalhauExecutor(options BacalhauExecutorOptions) (*BacalhauExecutor, error) {
+
+	apiHost := fmt.Sprintf("http://%s:%s", options.ApiHost, options.ApiPort)
+
+	client, err := newBacalhauClient(apiHost)
 	if err != nil {
 		return nil, err
 	}
 
-	apiHost := ""
-	if options.ApiHost != "DO_NOT_SET" {
-		apiHost = fmt.Sprintf("BACALHAU_API_HOST=%s", options.ApiHost)
-	}
-	bacalhauEnv := []string{
-		apiHost,
-		fmt.Sprintf("HOME=%s", os.Getenv("HOME")),
-	}
-	log.Debug().Msgf("bacalhauEnv: %s", bacalhauEnv)
-
 	return &BacalhauExecutor{
 		Options:        options,
-		bacalhauEnv:    bacalhauEnv,
 		bacalhauClient: *client,
-		ipfsClient:     ipfsClient,
 	}, nil
 }
 
 func (executor *BacalhauExecutor) Id() (string, error) {
-	nodeIdCmd := exec.Command(
-		"bacalhau",
-		"id",
-	)
-	nodeIdCmd.Env = executor.bacalhauEnv
-
-	runOutputRaw, err := nodeIdCmd.CombinedOutput()
-	if err != nil {
-		return "", fmt.Errorf("error calling get id results %s, %s", err.Error(), runOutputRaw)
-	}
-
-	splitOutputs := strings.Split(strings.Trim(string(runOutputRaw), " \t\n"), "\n")
-	runOutput := splitOutputs[len(splitOutputs)-1]
-
-	var idResult struct {
-		ID       string
-		ClientID string
-	}
-	err = json.Unmarshal([]byte(runOutput), &idResult)
+	id, err := executor.bacalhauClient.getID()
 	if err != nil {
-		return "", fmt.Errorf("error unmarshalling job JSON %s %s", err.Error(), runOutputRaw)
+		return "", fmt.Errorf("error getting bacalhau ID %s", err.Error())
 	}
-
-	return idResult.ID, nil
+	return id, nil
 }
 
 // Checks that Bacalhau is installed, correctly configured, and available
 func (executor *BacalhauExecutor) IsAvailable() (bool, error) {
-	isAlive, err := executor.bacalhauClient.alive()
-	if !isAlive || err != nil {
+	alive, err := executor.bacalhauClient.alive()
+	if !alive || err != nil {
 		return false, fmt.Errorf("Bacalhau is not currently available. Please ensure that Bacalhau is running, then try again. %w", err)
 	}
 
-	// Check that we have the right version of Bacalhau
 	version, err := executor.bacalhauClient.getVersion()
 	if err != nil {
-		return false, err
+		return false, fmt.Errorf("error getting bacalhau version %s", err.Error())
 	}
-	// TODO: we may want to relax this
-	if version.GitVersion != "v1.3.2" {
-		return false, errors.New("Bacalhau version must be v1.3.2")
+
+	if version < "v1.5.1" {
```

Comment:
Should we bump this check to `v1.6.x`? Or is `v1.5.1` still compatible?

Reply:
1.5.1 is still compatible (that's where we were when I picked up this PR)... I'm *pretty* sure this will go back to 1.4.x but haven't tested. I think we can leave it here

Reply:
Sounds good 👌